Scripts/zombie_head.py

```diff
@@ -0,0 +1,34 @@
+import pygame
+import time
+
+class ZombieHead(pygame.sprite.Sprite):
+    """
+    The ZombieHead class represents a zombie object in the game.
+    It can be spawned, smashed by the player, and automatically despawned after a set amount of time.
+    """
+
+    def __init__(self, spawn_data):
+        """
+        Initialize the ZombieHead with its image, sound, and spawn location.
+        """
+        super().__init__()
+        self.image = pygame.image.load('img/zombie_head.png')  # Optimized for performance with convert()
+        self.rect = self.image.get_rect()
```

Comment:
Kích thước hình ảnh lớn hơn rất nhiều so với cái đầu zombie, vì có những phần background trong suốt thừa ra bên ngoài. Nên xảy ra trường hợp người chơi ko cần bấm trúng vào đầu zombie mà vẫn được tính là hit.

Reply:
# Tạo mask cho collision detection
        self.mask = pygame.mask.from_surface(self.image)

        # Tạo một rect nhỏ hơn cho collision (ví dụ: 60% kích thước gốc)
        smaller_size = (int(self.rect.width * 0.6), int(self.rect.height * 0.6))
        self.collision_rect = pygame.Rect(0, 0, *smaller_size)
        self.collision_rect.center = self.rect.center
        
        
        thêm hàm check_hit 
        |def check_hit(self, pos):
        """
        Check if the given position hits the zombie.
        """
        offset_x = pos[0] - self.rect.x
        offset_y = pos[1] - self.rect.y
        if 0 <= offset_x < self.rect.width and 0 <= offset_y < self.rect.height:
            if self.mask.get_at((offset_x, offset_y)):
                return True

        return False